shell/platform/darwin/ios/ios_surface_metal_impeller.mm

```diff
@@ -58,8 +58,11 @@
 
   // When there are platform views in the scene, the drawable needs to be presented in the same
   // transaction as the one created for platform views. When the drawable are being presented from
-  // the raster thread, participate with the implicit CA transaction.
-  layer.presentsWithTransaction = YES;
+  // the raster thread, we may not be able to use a transaction as it will dirty the UIViews being
+  // presented. If there is a non-Flutter UIView active, such as in add2app or a
+  // presentViewController page transition, then this will cause CoreAnimation assertion errors and
+  // exit the app.
+  layer.presentsWithTransaction = [[NSThread currentThread] isMainThread];
```

Comment:
Wont' this re-introduce the bug we cherry picked a fix for into 3.10?

Reply:
Yup

Reply:
Is there any way we could add some tests in this? Explicitly start a transaction on the main thread, then try to commit one on the background?

Reply:
We can add tests in the framework repo. The assertion error cannot be triggered on Simulators IIRC